ddtrace/contrib/cassandra/session.py

```diff
@@ -31,10 +32,38 @@ def traced_connect(func, instance, args, kwargs):
     session = func(*args, **kwargs)
     if not isinstance(session.execute, wrapt.FunctionWrapper):
         # FIXME[matt] this should probably be private.
-        setattr(session, 'execute', wrapt.FunctionWrapper(session.execute, traced_execute))
+        setattr(session, 'execute_async', wrapt.FunctionWrapper(session.execute_async, traced_execute_async))
     return session
 
-def traced_execute(func, instance, args, kwargs):
+
+def traced_execute_async_callback(results, future, span):
+    span.set_tags(_extract_result_metas(cassandra.cluster.ResultSet(future, results)))
+    span.finish()
+
+def traced_execute_async_errback(exc, span):
+    # FIXME how should we handle an exception that hasn't be rethrown yet
+    try:
+        raise exc
```

Comment:
This is certainly not the right way to do it. Any idea, how do we usually handle this type of errors ?
BTW this means we will log the cassandra errors differently even for in the synchronous scenario.